src/display/ui_state.rs

```diff
@@ -30,6 +39,24 @@ impl Bandwidth for ConnectionData {
     fn get_total_bytes_downloaded(&self) -> u128 {
         self.total_bytes_downloaded
     }
+    fn get_avg_bytes_uploaded(&self) -> u128 {
```

Comment:
this code is repeated 4 times, does it make sense to refactor it to split the averaging logic from the access to the field?

Reply:
make sense, let me refactor that